frontend/src/helpers/utils.ts

```diff
@@ -82,3 +83,31 @@ export function parseJSONsafe(presumedJSON: string): any | null {
 export function formatDateToIsoString(dateToIsoString: Date) {
   return format(dateToIsoString, "yyyy-MM-dd");
 }
+export function calculatePrice(arrayItems: Pick<OfferItem, "amount" | "price">[], date?: string) {
+  let amountNet = 0;
+  let amountGross = 0;
+  let amountVat = 0;
+
+  for (let i = 0; i < arrayItems.length; i++) {
+    const amount = arrayItems[i].amount ?? 0;
+    const price = arrayItems[i].price ?? 0;
+
+    amountNet += round(amount * price, 2);
+    amountGross += round(amountNet * (1 + getVatRate({ isoDate: date })), 2);
+    amountVat = round(amountGross - amountNet, 2);
```

Comment:
This way probably makes more sense. 

It will also avoid a bug, which you have currently when showing the total sum of the offer

```suggestion
    amountNet += round(amount * price, 2);
    amountVat += round(amountNet * getVatRate({ isoDate: date }),2);
    amountGross += round(amountNet + amountVat, 2);
```

Reply:
You can keep the order like this of course. But then we need to make sure this bug is fixed

<img width="1439" alt="Screenshot 2024-08-15 at 06 58 52" src="https://github.com/user-attachments/assets/a0dd86c5-f538-43b5-b3ca-44491cd14d64">